### PR TITLE
Add decorator features

### PR DESCRIPTION
**Pull request by fernando-silvabr66**
Create a class Nameable.
Update person.rb file for Nameable inheritance.
Create a class Decorator.
Create a class Trimmer_Decorator.
Create a class Capitalize_Decorator.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,16 +1,23 @@
+require './nameable'
+
 # class for person data
 class Person
```
Comment (hiromataba) [inline person.rb]: - [x] Your Person class should inherit from the Nameable class

**Review by hiromataba — CHANGES_REQUESTED** [review]
Hi @fernando-silvabr66     :wave: :wave: 

almost-there

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Your PR has a proper title and description  :+1: 
- Your linters are passing :heavy_check_mark:  
- Profesional README file :rocket: 

## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (fernando-silvabr66) [reply person.rb]: Hi, thanks for the comment.
Just realize the file name was not correct.
Best regards.

Comment (hiromataba) [reply person.rb]: it is something like this :point_down: 
```ruby
class Person < Nameable
```

Comment (fernando-silvabr66) [reply person.rb]: Thanks... just fixed it.

File: person.rb [hunk 2]
```diff
@@ -1,16 +1,23 @@
+require './nameable'
+
 # class for person data
-class Person
+class Person < Nameable
```
Comment (hiromataba) [inline person.rb:4]: - [ ] Well done :wink:

**Review by hiromataba — APPROVED** [review]
# Status: Approved :tada: :tada: :tada: 
Hi @fernando-silvabr66     :wave: :wave: 

200w  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**